reframe/core/config.py

```diff
@@ -217,3 +220,121 @@ def create_env(system, partition, name):
                 system.add_partition(part)
 
             self._systems[sys_name] = system
+
+
+def convert_old_config(filename):
+    try:
+        old_config = load_settings_from_file(filename)
+    except (OSError, ReframeError) as e:
+        sys.stderr.write(
+            '%s: could not load settings: %s\n' % (sys.argv[0], e))
+        sys.exit(1)
+
+    if old_config is None:
+        raise ReframeFatalError('ReFrame is not configured')
+
+    converted = {
+        'systems': [],
+        'environments': [],
+        'logging': [],
+        'perf_logging': [],
+    }
+    old_systems = old_config.site_configuration['systems'].items()
+    for sys_name, sys_specs in old_systems:
+        sys_dict = {'name': sys_name}
+        sys_dict.update(sys_specs)
+        # Make variables dictionary into a list of lists
+        if 'variables' in sys_specs:
+            sys_dict['variables'] = []
```

Comment:
```suggestion
            sys_dict['variables'] = list(sys_specs['variables'].items())
```

Reply:
If the validation is not happy with the list of tuples, then just do it in a list comprehension as follows:

```suggestion
            sys_dict['variables'] = [[vname, v] for vname, v in sys_dict[ 'variables']]
```